upgrade-1.x-to-2.x.sh

```diff
@@ -0,0 +1,410 @@
+    psk=$(jq </mnt/boot/config.json '.["files"]."network/network.config"' | sed -e 's/.*Passphrase = \([^\\"]*\).*/\1/')
+
+    # Write NetworkManager setup
+    cat >/mnt/state/root-overlay/etc/NetworkManager/system-connections/resin-wifi <<EOF
+[connection]
+id=resin-wifi
+type=wifi
+
+[wifi]
+hidden=true
+mode=infrastructure
+ssid=$ssid
+
+[wifi-security]
+auth-alg=open
+key-mgmt=wpa-psk
+psk=$psk
+
+[ipv4]
+method=auto
+
+[ipv6]
+addr-gen-mode=stable-privacy
+method=auto
+EOF
+fi
+
+# Make /root/.docker
+mkdir -p /mnt/state/root-overlay/home/root/.docker
+
+# Touch openssl rnd file
+touch /mnt/state/root-overlay/home/root/.rnd
+
+# Restart docker
+systemctl start docker
+
+# Remount backup dir
+mount ${root_dev}p3 /tmp/backup
+
+IMAGE=resin/resinos:${TARGET_VERSION}-${DEVICE}
+CONTAINER=export-${TARGET_VERSION}
+BACKUPARCHIVE=/tmp/backup/newos.tar.gz
+FSARCHIVE=/mnt/data/newos.tar.gz
+
+# Create container for new version
+docker create --name ${CONTAINER} ${IMAGE} echo export
+
+# Export container
+docker export ${CONTAINER} | gzip > ${BACKUPARCHIVE}
+
+# Remove container
+docker rm ${CONTAINER}
+
+# Copy resin-data to backup partition before we wipe data
+cp /mnt/data/resin-data.tar.gz /tmp/backup/
+
+# Stop docker
+systemctl stop docker
+
+# Unmount data partition
+umount /mnt/data
+umount /var/lib/docker
+
+# Make ext4 data partition
+mkfs.ext4 -F -L resin-data ${root_dev}p6
+
+# Mount it
+mount ${root_dev}p6 /mnt/data
+
+# Copy resin-data backup and new OS to data partition
+(cd /mnt/data; tar xvf /tmp/backup/resin-data.tar.gz)
+cp ${BACKUPARCHIVE} ${FSARCHIVE}
+
+# Unmount backup dir
+umount /tmp/backup
+
+# Make new fs for rootB
+mkfs.ext4 -F -L resin-rootB ${root_dev}p3
+
+# Mount rootB partition
+mkdir -p /tmp/rootB
+mount ${root_dev}p3 /tmp/rootB
+
+# Extract rootfs to rootB
+echo quirks >/tmp/root-exclude
+echo resin-boot >>/tmp/root-exclude
+tar -x -X /tmp/root-exclude -C /tmp/rootB -f ${FSARCHIVE}
+
+# Extract quirks
+tar -x -C /tmp -f ${FSARCHIVE} quirks
+cp -a /tmp/quirks/* /tmp/rootB/
+rm -rf /tmp/quirks
+
+# XXX hack required until 2.0.7 when a quirk is added for this
+(cd /tmp/rootB/etc; rm mtab; ln -s ../proc/self/mounts mtab)
+
+# Unmount rootB partition
+umount /tmp/rootB
+
+# Extract boot partition, exclude boot_whitelist files
+echo resin-boot/cmdline.txt >/tmp/boot-exclude
+echo resin-boot/config.txt >>/tmp/boot-exclude
+echo resin-boot/splash/resin-logo.png >>/tmp/boot-exclude
+echo resin-boot/uEnv.txt >>/tmp/boot-exclude
+echo resin-boot/EFI/BOOT/grub.cfg >>/tmp/boot-exclude
+# 2.x adds a default config.json, we should avoid clobbering the existing one
+echo resin-boot/config.json >>/tmp/boot-exclude
+tar -x -X /tmp/boot-exclude -C /tmp -f ${FSARCHIVE} resin-boot
+cp -av /tmp/resin-boot/* /mnt/boot/
+
+# Remove OS image
+rm ${FSARCHIVE}
+
+# Switch root partition
+log "Switching root partition..."
+case $SLUG in
+    beaglebone*)
+	echo 'resin_root_part=3' >/mnt/boot/resinOS_uEnv.txt
+	rm /mnt/boot/uEnv.txt
```

Comment:
We should only remove the root partition variable definition if available and maybe mmc dev too (I can't remember exactly what was there before). But there can be things like fdtfile or similar which we don't want to remove as they can be added by the users for various reasons.